NPE from compiler instead of syntax error on `@anno`

Create following program:
```ruby
@anno
fn = 10
```
and execute it. The execution crashes with:
```
Caused by: org.graalvm.polyglot.PolyglotException: java.lang.NullPointerException: Cannot invoke "org.enso.compiler.core.ir.Expression.transformExpressions(scala.PartialFunction)" because "ir" is null
        at org.enso.runtime.compiler/org.enso.compiler.pass.desugar.SectionsToBinOp$.runExpression(SectionsToBinOp.scala:83)
        at org.enso.runtime.compiler/org.enso.compiler.pass.desugar.SectionsToBinOp$.$anonfun$runModule$1(SectionsToBinOp.scala:57)
        at org.enso.runtime.parser/org.enso.compiler.core.ir.Name$GenericAnnotation.mapExpressions(Name.scala:749)
        at org.enso.runtime.parser/org.enso.compiler.core.ir.Name$GenericAnnotation.mapExpressions(Name.scala:681)
        at org.enso.runtime.parser/org.enso.compiler.core.ir.Module.$anonfun$mapExpressions$3(Module.scala:144)
        at scala.library@2.13.11/scala.collection.immutable.List.map(List.scala:246)
        at org.enso.runtime.parser/org.enso.compiler.core.ir.Module.mapExpressions(Module.scala:144)
        at org.enso.runtime.compiler/org.enso.compiler.pass.desugar.SectionsToBinOp$.runModule(SectionsToBinOp.scala:55)
        at org.enso.runtime.compiler/org.enso.compiler.pass.PassManager.$anonfun$runPassesOnModule$2(PassManager.scala:101)

```

--- a/engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/ErrorCompilerTest.java
+++ b/engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/ErrorCompilerTest.java
@@ -25,6 +25,17 @@ public class ErrorCompilerTest extends CompilerTests {
   }
 
   @Test
+  public void brokenAnnotation() throws Exception {
+    var ir = parse("""
+    @anno
+    fn = 10
+    """);
+
+    assertSingleSyntaxError(
+        ir, Syntax.UnexpectedExpression$.MODULE$, "Unexpected expression", 0, 13);
+  }
+
+  @Test
   public void dotUnderscore() throws Exception {
     var ir = parse("""
     run op =

--- a/engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java
+++ b/engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java
@@ -4,6 +4,7 @@ import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Map;
 import java.util.UUID;
+
 import org.enso.compiler.core.ir.CallArgument;
 import org.enso.compiler.core.ir.DefinitionArgument;
 import org.enso.compiler.core.ir.Diagnostic;
@@ -42,6 +43,7 @@ import org.enso.syntax2.Token;
 import org.enso.syntax2.Tree;
 import org.enso.syntax2.Tree.Invalid;
 import org.enso.syntax2.Tree.Private;
+
 import scala.Option;
 import scala.collection.immutable.LinearSeq;
 import scala.collection.immutable.List;
@@ -318,10 +320,14 @@ final class TreeToIr {
       }
 
       case Tree.Annotated anno -> {
-        var annotationArgument = translateExpression(anno.getArgument());
-        var annotation = new Name.GenericAnnotation(anno.getAnnotation().codeRepr(),
-            annotationArgument, getIdentifiedLocation(anno), meta());
-        yield translateModuleSymbol(anno.getExpression(), join(annotation, appendTo));
+        if (anno.getArgument() == null) {
+          yield join(translateSyntaxError(anno, Syntax.UnexpectedExpression$.MODULE$), appendTo);
+        } else {
+          var annotationArgument = translateExpression(anno.getArgument());
+          var annotation = new Name.GenericAnnotation(anno.getAnnotation().codeRepr(),
+              annotationArgument, getIdentifiedLocation(anno), meta());
+          yield translateModuleSymbol(anno.getExpression(), join(annotation, appendTo));
+        }
       }
 
       case Tree.Documented doc -> {

